pkg/upgrade/upgrades/tenant_table_migration.go

```diff
@@ -80,6 +71,37 @@ SELECT n2.id
 	}
 
 	_, err = d.InternalExecutor.ExecEx(ctx, "add-system-entry", nil,
-		sessiondata.NodeUserSessionDataOverride, addSystemTenantEntry)
+		sessiondata.NodeUserSessionDataOverride,
+		`UPSERT INTO system.public.tenants (id, active, info, name, data_state, service_mode)
+VALUES (1, true,
+	crdb_internal.json_to_pb('cockroach.multitenant.TenantInfo', '{"deprecatedId":1,"deprecatedDataState":"READY"}'),
+  '`+catconstants.SystemTenantName+`',
+  `+strconv.Itoa(int(mtinfopb.DataStateReady))+`,
+  `+strconv.Itoa(int(mtinfopb.ServiceModeShared))+`)`,
+	)
 	return err
 }
+
+func getTenantsTableID(ctx context.Context, d upgrade.TenantDeps) (descpb.ID, error) {
```

Comment:
My sense is you could just as well use a testing knob and have the base case use the const. This is okay though.